src/flamegraph/mod.rs

```diff
@@ -254,15 +270,37 @@ where
             write!(buffer, "all ({} samples, 100%)", samples_txt)
         } else {
             let pct = (100 * samples) as f64 / timemax as f64;
-
-            // strip any annotation
-            write!(
-                buffer,
-                "{} ({} samples, {:.2}%)",
-                deannotate(&frame.location.function),
-                samples_txt,
-                pct
-            )
+            match frame.delta {
+                None => write!(
+                    buffer,
+                    "{} ({} samples, {:.2}%)",
+                    deannotate(&frame.location.function),
```

Comment:
Maybe hoist out `deannotate(&frame.location.function)` since it's used by all the arms?